crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -200,10 +203,90 @@ impl Feature<api::Authorize, types::PaymentsAuthorizeData> for types::PaymentsAu
                 &new_router_data.request,
                 &new_router_data.response,
             );
-
             new_router_data.integrity_check = integrity_result;
-
-            metrics::PAYMENT_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
+            // Check if the Capture API should be called based on the connector and other parameters
+            if helpers::should_call_capture_api(
+                &connector.connector_name,
+                self.request.off_session,
+                self.request.customer_acceptance,
+                self.request.capture_method,
+                self.request.setup_future_usage,
+                new_router_data.status,
+            ) {
+                // Convert Authorize RouterData into Capture RouterData
+                let mut capture_router_data = helpers::router_data_type_conversion(
+                    new_router_data.clone(),
+                    types::PaymentsCaptureData::foreign_try_from(new_router_data.clone())?,
+                    Err(types::ErrorResponse::default()),
+                );
+                let (connector_request, _) = capture_router_data
+                    .build_flow_specific_connector_request(
+                        state,
+                        connector,
+                        call_connector_action.clone(),
+                    )
+                    .await?;
+                let post_capture_router_data = capture_router_data
+                    .decide_flows(
+                        state,
+                        connector,
+                        call_connector_action,
+                        connector_request,
+                        business_profile,
+                        header_payload.clone(),
+                    )
+                    .await;
+                // Check if the post_capture_router_data is Ok or Err
+                match post_capture_router_data {
+                    Ok(post_capture_router_data) => {
+                        match (
+                            new_router_data.response.clone(),
+                            post_capture_router_data.response.clone(),
+                            post_capture_router_data.status,
+                        ) {
+                            (
+                                Ok(new_resp),
+                                Ok(post_capture_resp),
+                                common_enums::AttemptStatus::Charged,
+                            ) => {
+                                new_router_data.status = common_enums::AttemptStatus::Charged;
+                                new_router_data.response =
+                                    Ok(types::PaymentsResponseData::merge_transaction_responses(
+                                        &new_resp,
+                                        &post_capture_resp,
+                                    ));
+                            }
+                            (Err(new_resp_error), _, current_status) => {
```

Comment:
Why in all response we are not handing merge response? how status are handled in these cases?

Reply:
Removed the unnecessary branches